updater/lib/dependabot/job.rb

```diff
@@ -112,8 +112,9 @@ def vulnerable?(dependency)
         version_class_for_package_manager(dependency.package_manager)
       return false unless version_class.correct?(dependency.version)
 
-      version = version_class.new(dependency.version)
-      security_advisories.any? { |a| a.vulnerable?(version) }
+      all_versions = dependency.all_versions.
+                     filter_map { |v| version_class.new(v) if version_class.correct?(v) }
+      security_advisories.any? { |a| all_versions.any? { |v| a.vulnerable?(v) } }
```

Comment:
I've noticed this code

```ruby
version_class = Utils.version_class_for_package_manager(package_manager)
version_class.new(version) if version_class.correct?(version)
```

in a lot of places. Looking at it once again here, I'm wondering if it wouldn't make sense to put this in `Dependency`. It would be a reasonable argument to say it's too much for that class to know about, but it would remove some boilerplate. Maybe something like this (but probably with better naming)

```ruby
def correct_version
  version_class = Utils.version_class_for_package_manager(package_manager)
  version_class.new(version) if version_class.correct?(version)
end

def all_correct_versions
  version_class = Utils.version_class_for_package_manager(package_manager)
  all_versions.filter_map { |v| version_class.new(v) if version_class.correct?(v) }
end
```

Reply:
Yeah, that's a good thought! There's still some future changes needed to get a successful PR through. If that ends up duplicating this again I think that'd be good opportunity to dry it up?